.gitignore

```diff
@@ -10,4 +10,5 @@ bin
 out
 obj
 build
-TestResults
\ No newline at end of file
+TestResults
+.vscode
```

Comment:
Do we want to include vscode in the gitignore?

Reply:
I assumed we dont want this because anything we want to share about our dev setup lives in the devcontainer.json

I added an extension to my local setup that not everyone will want and this is where it showed up so I hid the directory to prevent it from ending up in source.

Reply:
@KanishkT123 any thoughts on how we should best manage the line between personal vscode and shared dev environment?

Reply:
But we would want to check in things like VSCode tasks for build and other actions so I do think the folder can exist. Compiled extensions *shouldn't* show up by default in the vscode folder. 

One idea we could do is lock the VSCode folder behind CODEOWNERS to make it harder for folks to accidentally check in an errant change. That would allow us to both have things like tasks.json AND prevent those kinds of weird checkins.